### PR TITLE
AllIndexController: capi 404 are now warning (not errors)

### PR DESCRIPTION
**Pull request by TBonnin**
## What does this change?
Logs capi 404 as warnings
I also reformat some of the code to make it more readable.

## What is the value of this and can you measure success?
less noise in kibana

## Tested in CODE?
Yes

### CODE REVIEW

File: applications/app/controllers/AllIndexController.scala [hunk 1]
```diff
@@ -33,9 +34,16 @@ class AllIndexController(contentApiClient: ContentApiClient, sectionsLookUp: Sec
...
+    lazy val fallbackToAll: Result = Found(s"/$path/all")
+    findByDate(path, reqDate)
+      .map { maybeDate =>
+        maybeDate
+          .map(date => Found(s"/$path/${urlFormat(date)}/all"))
+          .getOrElse(fallbackToAll)
+      }
+      .recover {
+        case _ => fallbackToAll
+      }
```
Comment (TBonnin) [inline applications/app/controllers/AllIndexController.scala]: formatting + adding recover

File: applications/app/controllers/AllIndexController.scala [hunk 2]
```diff
@@ -56,31 +64,39 @@ class AllIndexController(contentApiClient: ContentApiClient, sectionsLookUp: Sec
...
-        }
-        val today = DateTime.now
-        val nextPage = if (reqDate.sameDay(today)) None else Some(s"/$path/${urlFormat(reqDate.plusDays(1))}/altdate")
-        val model = index.copy(contents = contentOnRequestedDate, tzOverride = Some(DateTimeZone.UTC))
-
-        Cached(300)(RevalidatableResult.Ok(views.html.all(model, PreviousAndNext(prevPage, nextPage))))
+          .getOrElse(notFound)
+      }
+      .recover {
+        case _ => notFound
```
Comment (TBonnin) [inline applications/app/controllers/AllIndexController.scala]: formatting + adding recover

File: applications/app/controllers/AllIndexController.scala [hunk 3]
```diff
@@ -105,19 +121,32 @@ class AllIndexController(contentApiClient: ContentApiClient, sectionsLookUp: Sec
...
-      log.error(s"Cannot fetch content for path '$path'", e)
-      None
+    result.onFailure {
+      case GuardianContentApiError(404, _, _) =>
+        log.warn(s"Cannot fetch content for request '${request.uri}'")
+      case e: Exception =>
+        log.error(e.getMessage, e)
     }
+
+    result
```
Comment (TBonnin) [inline applications/app/controllers/AllIndexController.scala]: I dissociated logging and error handling.
I also move the error handling to the caller. Let me know how you feel about this

**Review by jfsoul — APPROVED** [review]
nice tidy 💎

**Pull request merged**